src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,318 @@
+store the user's credentials. In the long term this gives more flexibility as
+the users and keys can be authorized and revoked by the system administrator
+easily.
+
+Existing solution comparison
+****************************
+Existing Yubikey devices have already been enabled in RHEL and Fedora but the
+approach is completely different. The existing solution uses
+`OATH-TOTP <https://en.wikipedia.org/wiki/Time-based_one-time_password>`__ and
+`OATH-HOTP <https://en.wikipedia.org/wiki/HMAC-based_one-time_password>`__ to
+generate a one-time password (OTP) that is sent to the server as a second
+factor authentication. In other words, the authentication checks happen on the
+server. Moreover, this solution relies on a plugin for the 389 directory
+server, so it only works with IPA for IPA users.
+
+The approach explained in this document relies on the 
+`FIDO2 <https://en.wikipedia.org/wiki/FIDO2_Project>`__ standard, which uses a
+public-private key scheme to perform the authentication checks on the client.
+Besides, the solution is general enough to accept different LDAP servers like
+AD and IPA.
+
+Another difference is that with the current IPA feature, the first and second
+factor can also be used for doing a direct LDAP bind. This will not be possible
+with the new passkey scheme. Direct LDAP bind support will not be in the scope
+of this implementation.
+
+Implementation details
+----------------------
+
+The implementation will loosely follow existing smartcard authentication in
+SSSD. On top of that, and as mentioned before, the information will be stored
+in two places: in the LDAP server and on the passkey. The server will contain
+the public user credentials, including the user key mapping data, while the
+passkey device will store the private key necessary to authenticate the user.
+
+Key registration
+****************
+The first step is to register the user with a key. SSSD provides a helper
+utility to handle the passkeys, and it can be used to create the key mapping
+string (similar to pamu2fcfg). Instead of calling the helper utility from the
+command line the system administrator uses 
+`sssctl <https://docs.pagure.org/sssd.sssd/design_pages/sssctl.html>`__. This
+enables to unify the user experience. If sssctl is run by an unprivileged user,
+then additional udev rules might need to be configured. Please, check the
+instructions from the key provider to know how to add them.
+
+The output of this command can be set to the LDAP server in a dedicated
+attribute. For AD we use the
+`altSecurityIdentities <https://learn.microsoft.com/en-us/windows/win32/adschema/a-altsecurityidentities>`__
+with a prefix (i.e. passkey). For FreeIPA we use a new attribute called
+ipapasskey from the ipapasskeyuser objectclass. For any other LDAP server we
+use the passkey attribute.
+
+The format for
+`server-side credentials <https://developers.yubico.com/WebAuthn/WebAuthn_Developer_Guide/Resident_Keys.html>`__
```

Comment:
Are you sure this link is valid? It is the same as the one for discoverable credentials and does not explain server-side creds.